#!/usr/bin/env python3
# Dynamic reconfigure config for height PID controller
PACKAGE = "vpa_pi_drone"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("kp", double_t, 0, "Proportional gain", 15.0, 0.0, 100.0)
gen.add("ki", double_t, 0, "Integral gain", 2.0, 0.0, 50.0)
gen.add("kd", double_t, 0, "Derivative gain", 0.0, 0.0, 50.0)
gen.add("base_throttle", int_t, 0, "Base throttle for hover", 1500, 1000, 1800)
gen.add("height_ref", double_t, 0, "Target height (m)", 0.6, 0.0, 2.0)
gen.add("integral_limit", double_t, 0, "Integral clamp limit", 10.0, 0.0, 50.0)

exit(gen.generate(PACKAGE, "cfg", "HeightPID"))
